src/functional/ffront/type_info.py

```diff
@@ -350,8 +434,69 @@ def function_signature_incompatibilities(
             yield f"Expected keyword argument {kwarg} to be of type {func_type.kwargs[kwarg]}, but got {kwargs[kwarg]}."
 
 
+@function_signature_incompatibilities.register
+def function_signature_incompatibilities_fieldop(
+    func_type: ct.FieldOperatorType, args: list[ct.SymbolType], kwargs: dict[str, ct.SymbolType]
+) -> Iterator[str]:
+    yield from function_signature_incompatibilities_func(func_type.definition, args, kwargs)
+
+
+@function_signature_incompatibilities.register
+def function_signature_incompatibilities_scanop(
+    scanop_type: ct.ScanOperatorType, args: list[ct.SymbolType], kwargs: dict[str, ct.SymbolType]
+) -> Iterator[str]:
+    if not all(isinstance(arg, ct.FieldType) for arg in args):
+        yield "Arguments to scan operator must be fields."
+        return
+
+    arg_dims = [extract_dims(arg) for arg in args]
+    try:
+        promote_dims(*arg_dims)
+    except GTTypeError as e:
+        yield e.args[0]
+
+    if len(args) != len(scanop_type.definition.args) - 1:
+        yield f"Scan operator takes {len(scanop_type.definition.args)-1} arguments, but {len(args)} were given."
+        return
+
+    # build a function type to leverage the already existing signature checking
+    #  capabilities
+    function_type = ct.FunctionType(
+        args=[
+            ct.FieldType(dims=dims, dtype=dtype)
+            for dims, dtype in zip(arg_dims, scanop_type.definition.args[1:])
+        ],
+        kwargs={},
+        returns=ct.DeferredSymbolType(constraint=None),
+    )
+
+    yield from function_signature_incompatibilities(function_type, args, kwargs)
+
+
+@function_signature_incompatibilities.register
+def function_signature_incompatibilities_field(
+    field_type: ct.FieldType, args: list[ct.SymbolType], kwargs: dict[str, ct.SymbolType]
+) -> Iterator[str]:
+    if len(args) != 1:
+        yield f"Function takes 1 argument(s), but {len(args)} were given."
+        return
+
+    if not isinstance(args[0], ct.OffsetType):
+        yield f"Expected 0-th argument to be of type {ct.OffsetType}, but got {args[0]}."
+        return
+
+    if kwargs:
+        yield f"Got unexpected keyword argument(s) `{'`, `'.join(kwargs.keys())}`."
+        return
+
+    source_dim = args[0].source
+    target_dims = args[0].target
+    if field_type.dims and source_dim not in field_type.dims:
+        yield f"Incompatible offset can not shift field defined on " f"{', '.join([dim.value for dim in field_type.dims])} from " f"{source_dim.value} to target dim(s): " f"{', '.join([dim.value for dim in target_dims])}"
+
+
 def is_callable(
```

Comment:
I'm not to much into the `can_be` naming. How about `is_callable_with_args`?

Reply:
In my opinion, this function is not testing if the argument is a callable or a callable with some args; the function is checking if the `function_type` argument can be called with some specific args and kwargs (even the docstring mentions this: `Check if a function can be called for given arguments.`). I suggested `can_be_called` because it looks reasonable to me and also plays nicely with the name of the args (`with_args`, `with_kwargs`) but you're free to suggest a better name which is not misleading about the predicate that is checked here. Some other options I can think about are something like `accepts_args` or `support_args` ...